panther_analysis_tool/main.py

```diff
@@ -1106,124 +1106,94 @@ def _run_tests(
-        if auxiliary_functions_result_message:
-            print(auxiliary_functions_result_message)
+        spec = TestSpecification(
+            id=unit_test["Name"],
+            name=unit_test["Name"],
+            data=unit_test.get("Resource") or unit_test["Log"],
+            mocks=unit_test.get("Mocks", {}),
+            expectations=TestExpectations(detection=unit_test["ExpectedResult"]),
+        )
+        test_result = TestCaseEvaluator(spec, result).interpret()
+        test_result = _check_destinations(test_result, destination_by_name)
+        _print_test_result(test_result)
+        # add failed tests as necessary
+        if not test_result.passed:
+            failed_tests[detection.detection_id].append(unit_test["Name"])
 
     return failed_tests
 
 
-def _evaluate_auxiliary_function_result(
-    function_name: str, detection_result: DetectionResult, strict_check: bool
-) -> Dict[str, Any]:
-    """Determine whether an auxiliary function for a rule raised an error"""
-    function_error = getattr(detection_result, f"{function_name}_exception")
-    output = getattr(detection_result, f"{function_name}_output")
-    is_defined = getattr(detection_result, f"{function_name}_defined")
-    failed = function_error is not None
-
+def _check_destinations(test_result: TestResult, destination_by_name: Dict[str, Any]) -> TestResult:
     # For backwards compatibility we can accept invalid destination names,
     # as long as a string is returned. Strict check is enabled when users
     # pass destination names explicitly through command-line parameters.
-    if function_name == "destinations" and failed and not strict_check:
-        # Otherwise we fall back to a best-effort check of the return type only
-        if isinstance(function_error, UnknownDestinationError):
-            exc: UnknownDestinationError = function_error
-            failed = not _check_destinations_type(exc.result())
-
-    # The function has not been executed
-    if not is_defined:
-        display = None
-    else:
-        display = function_error if failed else output
-
-    return {
-        "error": function_error,
-        "output": output,
-        "failed": failed,
-        "has_invalid_type": isinstance(function_error, FunctionReturnTypeError),
-        "display": display,
-    }
-
-
-def _check_destinations_type(obj: Any) -> bool:
-    """Checks that the return value of the `destinations` function is a list of strings"""
-    if obj is None:
-        return True
-
-    if not isinstance(obj, list):
-        return False
+    if (
+        test_result.functions.destinationsFunction
+        and test_result.functions.destinationsFunction.error
+        and not bool(destination_by_name)
+    ):
 
-    return all(isinstance(destination_name, str) for destination_name in obj)
+        error_message = test_result.functions.destinationsFunction.error.message or ""
+        if "UnknownDestinationError" in error_message:
```

Comment:
This seems very hack-y, so definitely open to better way to support this backward compatibility.

Reply:
I don't think you can do this check from the `TestResult` attributes, you need access to `DetectionResult` which holds the exception instance.

Reply:
How about `TestCaseEvaluator.interpret` handled this internally? Perhaps adding an argument:

```python
def interpret(self, ignore_exception_types=None):
```

You can then check in `interpret` that an exception in a `DetectionResult` attribute (since you have access in there) is an instance of the given types and adjust the test result / function output accordingly.

How does that sound?

Reply:
Made a pass at this implementation. Didn't know if I should support passing the ignored exceptions all the way down the stack (perhaps sometime in the future we may support passing this in as a command line option?) - but I decided to do that anyway.